core/services/job/models.go

```diff
@@ -848,47 +849,50 @@ type LiquidityBalancerSpec struct {
 }
 
 type WorkflowSpec struct {
-	ID int32 `toml:"-"`
-	// TODO it may be possible to compute the workflow id from the hash(yaml, owner, name) and remove this field
-	WorkflowID    string    `toml:"workflowId"` // globally unique identifier for the workflow, specified by the user
-	Workflow      string    `toml:"workflow"`
-	WorkflowOwner string    `toml:"workflowOwner"` // hex string representation of 20 bytes
-	WorkflowName  string    `toml:"workflowName"`  // 10 byte plain text name
+	ID       int32  `toml:"-"`
+	Workflow string `toml:"workflow"` // the yaml representation of the workflow
+	// fields derived from the yaml spec, used for indexing the database
+	// note: i tried to make these private, but translating them to the database seems to require them to be public
+	WorkflowID    string    `toml:"-" db:"workflow_id"`    // Derived. Do not modify. the CID of the workflow.
+	WorkflowOwner string    `toml:"-" db:"workflow_owner"` // Derived. Do not modify. the owner of the workflow.
+	WorkflowName  string    `toml:"-" db:"workflow_name"`  // Derived. Do not modify. the name of the workflow.
```

Comment:
🤔 Are we removing the Owner + Name from the database / job spec? Does any functionality rely on the WorkflowOwner / WorkflowName being exposed at the DB level? (I'm thinking of CLO)

Reply:
yes, i'll have to update the RDD. 

yes, they will need to change some of there parsing.

they are using the `pkgworfklows.ParseWorkflowSpecYaml` today. they will be able to get these fields from the result of that, if they need them internally for their own private db purposes.

so instead of reading them from RDD (and forcing duplicated values between top-level RDD fields and embedded YAML tags), they can call a func they are already using and get the values